### ISSUE TITLE
azurerm_management_group_policy_remediation.policy_definition_reference_id should be case-insensitive

### ISSUE THREAD

**Issue by JustinGrote** [hashicorp/terraform-provider-azurerm#18846]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3

### AzureRM Provider Version

3.24.0

### Affected Resource(s)/Data Source(s)

azurerm_management_group_policy_remediation.policy_definition_reference_id

### Terraform Configuration Files

```hcl
resource "azurerm_management_group_policy_remediation" "inherit_tag_assignmentRemediation" {
  name                 = lower("${var.name}-inherit-tag1")
  management_group_id  = data.azurerm_management_group.target_management_group.id
  policy_assignment_id = azurerm_management_group_policy_assignment.inherit_tag_assignment.id
  policy_definition_reference_id = "/providers/Microsoft.Authorization/policyDefinitions/ea3f2387-9b95-492a-a190-fcdc54f7b070"
}
```

### Debug Output/Panic Output

```shell
None
```

### Expected Behaviour

Should Apply and be OK

### Actual Behaviour

Subsequent Applies don't recognize that Azure lowercased the reference ID even tho it is case insensitive:

```
      ~ policy_definition_reference_id = "/providers/microsoft.authorization/policydefinitions/ea3f2387-9b95-492a-a190-fcdc54f7b070" -> "/providers/Microsoft.Authorization/policyDefinitions/ea3f2387-9b95-492a-a190-fcdc54f7b070"
```

### References

Follow-on from #7018

**Comment by wuxu92** [comment]
Hi @JustinGrote thanks for your feedback. according to the API design, the `policy_definition_reference_id` should use the `reference_id` of the policy definition set not use the `policy definition id` directly. more information here <https://github.com/hashicorp/terraform-provider-azurerm/issues/17096#issuecomment-1151031270>. feel free to append to this thread if there are other questions.

**Comment by wuxu92** [comment]
@JustinGrote Sorry for my mistake. It still has a case-sensitivity problem use the reference_id. I am submitting a PR to fix this. but still you should use the reference id by discussions in the issues.

**Comment by JustinGrote** [comment]
> @JustinGrote Sorry for my mistake. It still has a case-sensitivity problem use the reference_id. I am submitting a PR to fix this. but still you should use the reference id by discussions in the issues.

Yes I used a bad example, but it is case sensitive when it shouldn't be even for reference id. Thanks!
